File: memory-domain/src/main/java/com/memory/domain/game/repository/GameSettingRepositoryCustom.java
package com.memory.domain.game.repository;

import com.memory.domain.game.GameMode;
import com.memory.domain.game.GameSetting;

import java.util.Optional;

public interface GameSettingRepositoryCustom {

    Optional<GameSetting> findByGameModeAndIsActiveTrue(GameMode gameMode);

}

File: memory-domain/src/main/kotlin/com/memory/domain/game/repository/GameSettingRepositoryCustom.kt
package com.memory.domain.game.repository

import com.memory.domain.game.GameMode
import com.memory.domain.game.GameSetting
import java.util.Optional

interface GameSettingRepositoryCustom {

    fun findByGameModeAndIsActiveTrue(gameMode: GameMode?): Optional<GameSetting>

}
